crates/storage/src/connection/transaction.rs

```diff
@@ -57,12 +68,49 @@ pub(super) fn insert_transactions(
 
     let events = transaction_data
         .iter()
-        .flat_map(|(_, receipt)| &receipt.events);
+        .filter_map(|(_, receipt)| receipt.as_ref())
+        .flat_map(|receipt| &receipt.events);
     super::event::insert_block_events(tx, block_number, events)
         .context("Inserting events into Bloom filter")?;
     Ok(())
 }
 
+pub(super) fn update_receipt(
+    tx: &Transaction<'_>,
+    block_hash: BlockHash,
+    transaction_idx: usize,
+    receipt: &Receipt,
+) -> anyhow::Result<()> {
+    let mut compressor = zstd::bulk::Compressor::new(10).context("Create zstd compressor")?;
+    let receipt = dto::Receipt::from(receipt);
+    let serialized_receipt = serde_json::to_vec(&receipt).context("Serializing receipt")?;
+    let serialized_receipt = compressor
+        .compress(&serialized_receipt)
+        .context("Compressing receipt")?;
+
+    let execution_status = match receipt.execution_status {
+        dto::ExecutionStatus::Succeeded => 0,
+        dto::ExecutionStatus::Reverted { .. } => 1,
+    };
+
+    tx.inner()
+        .execute(
+            r"
+            UPDATE starknet_transactions SET receipt = :receipt, execution_status = :execution_status
+            WHERE block_hash = :block_hash AND idx = :idx;
+            ",
+            named_params![
+                ":receipt": &serialized_receipt,
+                ":execution_status": &execution_status,
+                ":block_hash": &block_hash,
+                ":idx": &transaction_idx.try_into_sql_int()?,
+            ],
+        )
+        .context("Inserting transaction data")?;
+
```

Comment:
Seems like we're missing updating the events Bloom filter when using `update_receipt()` to add receipts to transactions. Should be added to a place where we have _all_ receipt for the block.